### ISSUE TITLE
Rails 5.2 :`FrozenError: can't modify frozen ActiveSupport::HashWithIndifferentAccess`

### ISSUE THREAD

**Issue by yahonda** [sportngin/active_zuora#85]
### Steps to reproduce

1. Install Ruby 2.4.6 (or whatever Ruby version Rails 5.2 supports like Ruby 2.2+)
2. export ZUORA_USER=<zuora_username> ; export ZUORA_PASS=<zuora_password> to run Integration tests
3. git clone https://github.com/sportngin/active_zuora
4. cd active_zuora
5. bundle install
6. bundle exec rspec

### Expected behavior
All of integration tests pass.

### Actual behavior

8 specs fail due to ` FrozenError: can't modify frozen ActiveSupport::HashWithIndifferentAccess`. I think it is related with `# frozen_string_literal: true` magic comment introduced since Rails 5.2.
https://github.com/rails/rails/blob/v5.2.3/activesupport/lib/active_support/hash_with_indifferent_access.rb#L1

```ruby
$ bundle exec rspec

Create Account
  can create an account (FAILED - 1)

ActiveZuora::Base
  #fields_order
    When the value of a field is null, it should be the first
    When the field name is id, it should be after the nil value fields but before all other fields

BelongsToAssociations
  should define a attribute assignment method method for the object
  should define a attribute assignment method for the object id

ActiveZuora::CollectionProxy
  should initialize
  should respond to enumerable methods
  should respond to the batch_subscribe method

ActiveZuora::Connection
  custom header
    passes the regular header if not set
    merges in a custom header if set
  login
    when a custom header is set
      uses the custom header
    when a custom header is not set
      does not use the custom header

ActiveZuora::DateField
  #type_cast
    returns nil if provided nil
    returns a date when given a datetime object
    returns a date when given a date object
  #build_xml
    handles a nil value
    handles a date value

HasManyRelations
  can specify conditions and order (FAILED - 2)
  can behave like an array (FAILED - 3)
  can respond to functions on the Relation (FAILED - 4)

LazyAttr
  should fetch a lazy loaded attribute from the api
  should should not refetch an attribute after it's been loaded once

Subscribe
  Can successfully subscribe and amend using a new account (FAILED - 5)
  Can successfully subscribe and generate a bill preview and an invoice (FAILED - 6)
  Can successfully batch subscribe from a collection proxy of subscribe requests and generate an invoice for the first subscription (FAILED - 7)

ZObject
  can be created, queried, updated, and destroyed (FAILED - 8)

Failures:

  1) Create Account can create an account
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./spec/account_integration_spec.rb:30:in `block (4 levels) in <top (required)>'
     # ./spec/spec_helper.rb:27:in `now'
     # ./spec/account_integration_spec.rb:29:in `block (3 levels) in <top (required)>'

  2) HasManyRelations can specify conditions and order
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./lib/active_zuora/persistence.rb:17:in `save!'
     # ./lib/active_zuora/persistence.rb:79:in `tap'
     # ./lib/active_zuora/persistence.rb:79:in `create!'
     # ./spec/has_many_integration_spec.rb:8:in `block (3 levels) in <top (required)>'

  3) HasManyRelations can behave like an array
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./lib/active_zuora/persistence.rb:17:in `save!'
     # ./lib/active_zuora/persistence.rb:79:in `tap'
     # ./lib/active_zuora/persistence.rb:79:in `create!'
     # ./spec/has_many_integration_spec.rb:8:in `block (3 levels) in <top (required)>'

  4) HasManyRelations can respond to functions on the Relation
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./lib/active_zuora/persistence.rb:17:in `save!'
     # ./lib/active_zuora/persistence.rb:79:in `tap'
     # ./lib/active_zuora/persistence.rb:79:in `create!'
     # ./spec/has_many_integration_spec.rb:8:in `block (3 levels) in <top (required)>'

  5) Subscribe Can successfully subscribe and amend using a new account
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./lib/active_zuora/persistence.rb:17:in `save!'
     # ./lib/active_zuora/persistence.rb:79:in `tap'
     # ./lib/active_zuora/persistence.rb:79:in `create!'
     # ./spec/subscribe_integration_spec.rb:8:in `block (3 levels) in <top (required)>'

  6) Subscribe Can successfully subscribe and generate a bill preview and an invoice
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./lib/active_zuora/persistence.rb:17:in `save!'
     # ./lib/active_zuora/persistence.rb:79:in `tap'
     # ./lib/active_zuora/persistence.rb:79:in `create!'
     # ./spec/subscribe_integration_spec.rb:8:in `block (3 levels) in <top (required)>'

  7) Subscribe Can successfully batch subscribe from a collection proxy of subscribe requests and generate an invoice for the first subscription
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./lib/active_zuora/persistence.rb:17:in `save!'
     # ./lib/active_zuora/persistence.rb:79:in `tap'
     # ./lib/active_zuora/persistence.rb:79:in `create!'
     # ./spec/subscribe_integration_spec.rb:8:in `block (3 levels) in <top (required)>'

  8) ZObject can be created, queried, updated, and destroyed
     Failure/Error: changed_attributes.clear

     FrozenError:
       can't modify frozen ActiveSupport::HashWithIndifferentAccess
     # ./lib/active_zuora/fields.rb:55:in `clear'
     # ./lib/active_zuora/fields.rb:55:in `clear_changed_attributes'
     # ./lib/active_zuora/persistence.rb:59:in `create'
     # ./lib/active_zuora/persistence.rb:13:in `save'
     # ./spec/zobject_integration_spec.rb:27:in `block (3 levels) in <top (required)>'

Finished in 4.23 seconds (files took 1.26 seconds to load)
26 examples, 8 failures

Failed examples:

rspec ./spec/account_integration_spec.rb:17 # Create Account can create an account
rspec ./spec/has_many_integration_spec.rb:28 # HasManyRelations can specify conditions and order
rspec ./spec/has_many_integration_spec.rb:37 # HasManyRelations can behave like an array
rspec ./spec/has_many_integration_spec.rb:42 # HasManyRelations can respond to functions on the Relation
rspec ./spec/subscribe_integration_spec.rb:41 # Subscribe Can successfully subscribe and amend using a new account
rspec ./spec/subscribe_integration_spec.rb:124 # Subscribe Can successfully subscribe and generate a bill preview and an invoice
rspec ./spec/subscribe_integration_spec.rb:228 # Subscribe Can successfully batch subscribe from a collection proxy of subscribe requests and generate an invoice for the first subscription
rspec ./spec/zobject_integration_spec.rb:13 # ZObject can be created, queried, updated, and destroyed

$
```

**Comment by yahonda** [comment]
I have confirmed this issue can be addressed by #84.

**Comment by yahonda** [comment]
Let me close this issue because this is kind of duplicate with #84.
